src/tools/db-upgrade/DbUpgraderTool.cpp

```diff
@@ -164,15 +174,96 @@ int main(int argc, char* argv[]) {
+               << " illegal, raw_data_version can only be 1 or 2";
     return EXIT_FAILURE;
   }
-  LOG(INFO) << "Prepare phase end";
+
+  // one space, check partId
+  std::vector<std::string> parts;
+  std::unordered_set<PartitionID> partsAll;
+  folly::split(",", FLAGS_partIds, parts, true);
+  for (auto& part : parts) {
+    PartitionID partId;
+    try {
+      partId = folly::to<PartitionID>(part);
+    } catch (const std::exception& ex) {
+      LOG(ERROR) << "Cannot convert part id " << part;
+      return EXIT_FAILURE;
+    }
+    partsAll.emplace(partId);
+  }
+
+  for (auto& path : srcPaths) {
+    auto dataPath = nebula::fs::FileUtils::joinPath(path, "nebula");
+    if (!nebula::fs::FileUtils::exist(dataPath)) {
+      LOG(ERROR) << "Source data path " << dataPath << " not exists!";
+      return EXIT_FAILURE;
+    }
+    auto subDirs = nebula::fs::FileUtils::listAllDirsInDir(dataPath.c_str());
+    if (subDirs.size() != 1) {
+      LOG(ERROR) << "Only handle one space!";
+      return EXIT_FAILURE;
+    }
+    auto space = subDirs[0];
+    GraphSpaceID spaceId;
+    try {
+      spaceId = folly::to<GraphSpaceID>(space);
+    } catch (const std::exception& ex) {
+      LOG(ERROR) << "Cannot convert space id " << space;
+      return EXIT_FAILURE;
+    }
+
+    auto sRet = schemaMan->toGraphSpaceName(spaceId);
+    if (!sRet.ok()) {
+      LOG(ERROR) << "Space id " << spaceId << " no found";
+      return EXIT_FAILURE;
+    }
+
+    auto spaceVidLenRet = metaClient->getSpaceVidLen(spaceId);
+    if (!spaceVidLenRet.ok()) {
+      LOG(ERROR) << "Space id " << spaceId << " vid len no found";
+      return EXIT_FAILURE;
+    }
+    auto spaceVidLen = spaceVidLenRet.value();
+    std::unique_ptr<nebula::kvstore::RocksEngine> readEngine(
+        new nebula::kvstore::RocksEngine(spaceId, spaceVidLen, path, "", nullptr, nullptr, true));
+
+    auto prefix = nebula::NebulaKeyUtilsV1::systemPrefix();
+    std::unique_ptr<nebula::kvstore::KVIterator> iter;
+    auto retCode = readEngine->prefix(prefix, &iter);
+    if (retCode != nebula::cpp2::ErrorCode::SUCCEEDED) {
+      LOG(ERROR) << "Space id " << spaceId << " get part system data failed";
+      return EXIT_FAILURE;
+    }
+
+    while (iter && iter->valid()) {
+      auto key = iter->key();
+      auto partId = nebula::NebulaKeyUtilsV1::getPart(key);
+      if (nebula::NebulaKeyUtilsV1::isSystemCommit(key)) {
+        // do nothing
+      } else if (nebula::NebulaKeyUtilsV1::isSystemPart(key)) {
+        LOG(INFO) << "Find part id " << partId << " system data.";
+        partsAll.erase(partId);
+      } else {
+        LOG(FATAL) << "Space id " << spaceId << " part " << partId << " illegal data!";
+      }
+      iter->next();
+    }
+  }
+  if (!partsAll.empty()) {
+    std::string partstr;
+    for (auto part : partsAll) {
+      partstr += folly::stringPrintf("%d ", part);
+    }
+    LOG(FATAL) << "There are some part " << partstr << "not found";
```

Comment:
Good! Just use `LOG(ERROR)` and `return EXIT_FAILURE;`, no need to core.

Reply:
ok